pilot/cmd/pilot-discovery/main.go

```diff
@@ -144,9 +144,8 @@ func init() {
 		"Setting this flag has no effect. Install CRD definitions directly or with the operator")
 	discoveryCmd.PersistentFlags().StringVar(&serverArgs.Config.FileDir, "configDir", "",
 		"Directory to watch for updates to config yaml files. If specified, the files will be used as the source of config, rather than a CRD client.")
-	discoveryCmd.PersistentFlags().StringVarP(&serverArgs.Config.ControllerOptions.WatchedNamespace, "appNamespace",
-		"a", metav1.NamespaceAll,
-		"Restrict the applications namespace the controller manages; if not set, controller watches all namespaces")
+	discoveryCmd.PersistentFlags().StringVarP(&serverArgs.Config.ControllerOptions.WatchedNamespaces, "appNamespace", "a", metav1.NamespaceAll,
```

Comment:
Please deprecate this option. Use MeshConfig !

Reply:
Planning on doing that next. Just didn't want to make this PR any larger than necessary.